pkg/object/object.go

```diff
@@ -50,14 +73,135 @@ type OpenCompose struct {
 	Volumes  []Volume
 }
 
+// Given the name of 'emptyDirVolume' this function searches
+// if service receiver has that 'emptyDirVolume'
+func (s *Service) EmptyDirVolumeExists(name string) bool {
+	for _, emptyDirVolume := range s.EmptyDirVolumes {
+		if name == emptyDirVolume.Name {
+			return true
+		}
+	}
+	return false
+}
+
+// Given name of root level 'volume' this function searches
+// if opencompose receiver has that 'volume'.
+func (o *OpenCompose) VolumeExists(name string) bool {
+	for _, volume := range o.Volumes {
+		if name == volume.Name {
+			return true
+		}
+	}
+	return false
+}
+
+// Documentation about the valid identifiers can be found at
+// https://github.com/kubernetes/community/blob/master/contributors/design-proposals/identifiers.md
+func validateName(name string) error {
+	if errs := validation.IsDNS1123Subdomain(name); len(errs) != 0 {
+		return fmt.Errorf("%s", strings.Join(errs, "\n"))
+	}
+	return nil
+}
+
+func validatePath(path string) error {
+	// TODO: implement me
+	return nil
+}
+
+func (c *Container) Validate() error {
+
+	// validate image name
+	// TODO: implement me
+	// validate Environment
+	// TODO: implement me
+	// validate Ports
+	// TODO: implement me
+
+	allMounts := make(map[string]string)
+	// validate Mounts
+	for _, mount := range c.Mounts {
+		// validate volumeName
```

Comment:
The problem is that we might not be able to do the validation of colliding `mountPath` ? Since the variable `allMounts` is defined outside the for loop and used inside the loop, one workaround would be to pass it with every call but then it makes mount's `validate` function inconsistent than other `validate` functions.